src/storage/txn/flow_controller/singleton_flow_controller.rs

```diff
@@ -728,13 +728,16 @@ impl<E: FlowControlFactorStore + Send + 'static> FlowChecker<E> {
         // Because pending compaction bytes changes dramatically, take the
         // logarithm of pending compaction bytes to make the values fall into
         // a relative small range
-        let num = (self.engine.pending_compaction_bytes(self.region_id, &cf) as f64).log2();
+        let mut num = (self.engine.pending_compaction_bytes(self.region_id, &cf) as f64).log2();
+        if !num.is_finite() {
+            // 0.log2() == -inf, which is not expected
+            num = 0.0;
+        }
         let checker = self.cf_checkers.get_mut(&cf).unwrap();
         checker.long_term_pending_bytes.observe(num);
```

Comment:
BTW, when `num` is 0, should we treat it as an invalid value and do not count it in `long_term_pending_bytes`.

Reply:
I think it's okay. 0 is valid representing no need to compact, should count in `long_term_pending_bytes`